src/extension.js

```diff
@@ -80,17 +98,95 @@ function displaySearchResults(results) {
 	}
 }
 
+function updateStatusBarItem(statusBarItem, uri) {
+
+	if (uri) {
+		currentProject = fileProvider.getProject(uri.path);
+		if (currentProject) {
+			const projectUri = fileProvider.getProjectUri(currentProject);
+			const data = projectDataProvider.getWorkspaceData(projectUri.path);
+			if (data) {
+				statusBarItem.text = `$(git-branch) ${currentProject.name}: ${data.branch}`;
+				statusBarItem.show();
+			}
+		}
+	} else {
+		statusBarItem.hide();
+	}
+}
+
+async function switchBranch(project) {
+	if (!project) {
+		return;
+	}
+
+	try {
+		const branches = await fileProvider.fetchBranches(project);
+
+		const prevBranch = project.branchId ? (branches.find( b => b.id === project.branchId )).name : 'main';
+		const names = branches.map(branch => branch.name );
+
+		// put current branch on the first place in the list
+		const index = names.indexOf(prevBranch);
+		if (index !== -1) {
+			names.splice(index, 1);
+			names.unshift(prevBranch);
+		}
+
+		const branch = await vscode.window.showQuickPick(names, { placeHolder: 'Select a branch to switch to' });
+
+		if (branch) {
+
+			// switch to the selected branch
+			if (prevBranch !== branch) {
+				fileProvider.switchBranch(project, branch);
+				const projectUri = fileProvider.getProjectUri(project);
+				projectDataProvider.setWorkspaceData(projectUri.path, { 
+					branch: branch
+				});
+				vscode.window.showInformationMessage(`Switched to branch ${branch}`);
+				await fileProvider.refreshProject(project);
+
+				// update status bar
+				updateStatusBarItem(statusBar, projectUri);
+
+				vscode.commands.executeCommand('workbench.action.closeAllEditors');
+				vscode.commands.executeCommand('workbench.files.action.refreshFilesExplorer');
+			}
+		}
+	} catch (error) {
+		console.error('Error during switching branches:', error);
+	}
+}
+
 /**
  * @param {vscode.ExtensionContext} context
  */
 async function activate(context) {
 
+	projectDataProvider = new ProjectDataProvider(context);
+	vscode.window.registerTreeDataProvider('workspaceView', projectDataProvider);
+
 	if (!fileProvider) {
-		fileProvider = new CloudStorageProvider(context);
+		fileProvider = new CloudStorageProvider(context, projectDataProvider);
 	}
-	
+
 	context.subscriptions.push(vscode.workspace.registerFileSystemProvider('playcanvas', fileProvider, { isCaseSensitive: true }));
 
+	// display the current branch in the status bar
+	statusBar = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left, 100);
+	statusBar.command = 'playcanvas.switchBranch';
+	context.subscriptions.push(statusBar);
+  
+	updateStatusBarItem(statusBar);
+
+    // Update status bar on editor change
+    context.subscriptions.push(vscode.window.onDidChangeActiveTextEditor(editor => {
+		if (editor && editor.document) {
+        	updateStatusBarItem(statusBar, editor.document.uri);
+		}
+    }));
```

Comment:
I can see a mix of tabs and spaces used for indentation. Probably a good idea to enable Prettier or ESLint at some point.

Reply:
i run clang format for all files